src/peft/tuners/lora/model.py

```diff
@@ -556,10 +569,16 @@ def add_weighted_adapter(
                         elif adapter in target.lora_embedding_A:
                             current_adapter_lora_A = target.lora_embedding_A[adapter]
                             current_adapter_lora_B = target.lora_embedding_B[adapter]
+                        else:
+                            continue
                         loras_A.append(current_adapter_lora_A.data * weight * target.scaling[adapter])
                         loras_B.append(current_adapter_lora_B.data)
-                    torch.cat(loras_A, dim=0, out=target_lora_A.data)
-                    torch.cat(loras_B, dim=1, out=target_lora_B.data)
+
+                    if len(loras_A) > 0:
```

Comment:
Yes, let's same there are 5 adapters, however we are combining only 2 adapters and if those 2 adapters don't target a specific layer while other 3 do, then it is fine to not do anything for this layer

Reply:
Wait a second, there will never be a case when no adapter is matching because the target modules are based on the adapters being combined. Therefore, at least there will be 1 matching adapter.

Reply:
So technically, `len(loras_A)` should never be 0, right? We could raise an error if it is with a message to report a bug on GH.

Reply:
addressed in the latest commit